src/main/java/org/assertj/core/util/Strings.java

```diff
@@ -83,7 +83,8 @@ public static String concat(Object... objects) {
    * @return the formatted string if any args were given
    */
   public static String formatIfArgs(String message, Object... args) {
-    return Arrays.isNullOrEmpty(args) ? message : format(message, args);
+    return Arrays.isNullOrEmpty(args) ? format(message.replaceAll("%","%%").replaceAll("%%n","%n")).intern() : format
```

Comment:
I don't think we need to intern the String, if the reason to do that was AbstractAssert_describedAs_with_text_description_Test.should_set_description failure, we should simply change the test and use `assertThat(assertions.descriptionText()).isEqualTo(description);` instead checking for String to be the same.
A comment would be welcome to explain the trick with % replacements.